frontend/src/app/features/map/search/RadiusSearch.tsx

```diff
@@ -0,0 +1,95 @@
+import { Button, Slider, Typography } from '@mui/material';
+import clsx from 'clsx';
+import { ActiveToolEnum, MIN_CIRCLE_RADIUS } from '../../../constants/Constant';
+import { XmarkIcon } from '../../../components/common/icon';
+import DropdownButton from '../DropDownButton';
+import { formatDistance } from '../../../helpers/utility';
+import { useState } from 'react';
+
+interface Props {
+  isSmall?: boolean;
+  className?: string;
+  setActiveTool?: React.Dispatch<React.SetStateAction<ActiveToolEnum | null>>;
+}
+
+export function RadiusSearch({
+  isSmall = false,
+  className,
+  setActiveTool,
+}: Readonly<Props>) {
+  const [radius, setRadius] = useState(MIN_CIRCLE_RADIUS);
+
+  const [isVisible, setIsVisible] = useState(true);
+  const onCancel = () => {
+    setIsVisible(false);
+    setActiveTool && setActiveTool(null);
+  };
+
+  const onRadiusChange = (_ev: any, value: number | number[]) => {
+    const newRadius = Math.max(
+      Array.isArray(value) ? value[0] : value,
+      MIN_CIRCLE_RADIUS,
+    );
+    setRadius(newRadius);
+  };
+
+  const sliderBox = (
+    <div className="point-search-slider-content">
+      {isSmall && (
+        <Typography className="point-search-slider-text">
+          Set Radius:
+        </Typography>
+      )}
+      <Slider
+        className={clsx(
+          'point-search-slider',
+          isSmall && 'point-search-slider--shrink',
+        )}
+        aria-label="Search radius"
+        valueLabelDisplay="off"
+        min={MIN_CIRCLE_RADIUS}
+        // 500 km is roughly half the size of BC
+        max={500000}
+        step={MIN_CIRCLE_RADIUS}
+        defaultValue={MIN_CIRCLE_RADIUS}
+        value={radius}
+        onChange={onRadiusChange}
+      />
+      <Typography className="point-search-slider-text">
+        {formatDistance(radius, 1)}
+      </Typography>
+    </div>
+  );
+
+  return isSmall ? (
+    sliderBox
+  ) : (
+    <div className={clsx('point-search', className)}>
+      {isVisible && (
+        <>
+          <Button
+            color="primary"
+            variant="contained"
+            size="medium"
+            onClick={onCancel}
+            className="radius-search-cancel-button"
+            startIcon={<XmarkIcon className="point-cancel-icon" />}
+          >
```

Comment:
Same comment as above: consider using the shared component that is consistent with the design system. This should also eliminate the need for the `.radius-search-cancel-button` and `.radius-search-set-radius` classes

Reply:
Will go ahead with Button coming from frontend/src/app/components/button/Button.tsx. Thanks for the feedback.